.github/pull_request_template.md

```diff
@@ -0,0 +1,24 @@
+# Pull Request Template
+
+## PR Title
+[Provide a short and descriptive title for the pull request, e.g., "Improve caching mechanism for API calls"]
+
+## Type of Change (Mandatory)
```

Comment:
This is a good improvement on the first draft.

My suggestion would be to remove this checkbox section. It helps in two ways:
1. Makes the template easier to approach (less overwhelming) and fits in a single screen. 
2. Description generally obviates the need for "Type of change" selection.